app/src/main/java/kumamotone/sample/eremotethingssample/MainActivity.kt

```diff
@@ -16,8 +17,8 @@ class MainActivity : Activity() {
     private fun String.hexStringToByteArray(): ByteArray =
             ByteArray(this.length / 2) { this.substring(it * 2, it * 2 + 2).toInt(16).toByte() }
 
-    private fun sendToRemocon(data: String) {
-        launch {
+    private suspend fun sendToRemocon(data: String) {
+        withContext(CommonPool) {
```

Comment:
`withContext` is a suspending function to switch the current coroutine context.
In this situation, we want to call `sendToRemoconWithoutCoroutine` in worker thread, so I use `withContext(CommonPool)` .

Reply:
I think I probably understood that now... We needed to use `withContext` to make this function suspend, right? I assume this is the point of my misunderstanding.

Reply:
Exactly. We have two options to make functions invoking on worker thread:

1. suspending function style by `withContext`
2. kick a new coroutine by `launch` or `async`

We should prefer to use the suspending function style 👍